### PR TITLE
Add flexible I-SUPPORT PCS segmentation

### LINKED ISSUE #100: PCS with rigid link and fixed joint
Add rigid link with fixed joint in the `ISupport` class. I tried with this draft [here](https://github.com/tud-phi/soromox/blob/amisupport_with_interfaces/src/soromox/systems/pcs/pcs.py). Still have to check it.

### PR DESCRIPTION
**Pull request by mstoelzle**
## Summary

This PR adds native I-SUPPORT support for rigid base/interface/tip connectors and flexible PCS discretization of each physical pneumatic segment.

Key API split:
- `ISupportParams` owns editable metric parameters, including pneumatic segment `length`, optional flattened `pcs_segment_lengths`, and optional `rigid_connector_lengths`.
- `ISupportStructure` owns fixed topology, including `pcs_segment_counts` and `rigid_connector_selector`.
- Pneumatic segment actuation remains grouped by physical pneumatic segment, so subdividing one pneumatic segment into multiple PCS segments does not increase the number of actuators.

## Motivation

- Native support for rigid interfaces: model the rigid base section, rigid interfaces between pneumatic segments, and the rigid tip directly as inactive PCS segments.
- More flexible PCS segmentation: divide each pneumatic segment into any number of PCS segments while keeping the pressure input layout unchanged.

## API Usage Examples

Default I-SUPPORT layout: one PCS segment per pneumatic segment, no rigid connectors.

```python
params = ISupportParams(
    ...,
    length=jnp.array([190e-3, 180e-3]),
)
robot = ISupport(params=params, structure=ISupportStructure())
```

Equal PCS subdivision of each pneumatic segment. Omitting `pcs_segment_lengths` makes each pneumatic segment split evenly according to `pcs_segment_counts`.

```python
params = ISupportParams(
    ...,
    length=jnp.array([190e-3, 180e-3]),
)
structure = ISupportStructure(pcs_segment_counts=(2, 3))
robot = ISupport(params=params, structure=structure)
```

Explicit PCS segment lengths. Lengths are flattened by pneumatic segment and each group must sum to the matching `params.length` entry.

```python
params = ISupportParams(
    ...,
    length=jnp.array([190e-3, 180e-3]),
    pcs_segment_lengths=jnp.array([
        95e-3, 95e-3,          # pneumatic_segment[0]
        60e-3, 60e-3, 60e-3,   # pneumatic_segment[1]
    ]),
)
structure = ISupportStructure(pcs_segment_counts=(2, 3))
robot = ISupport(params=params, structure=structure)
```

Rigid connectors use the canonical I-SUPPORT connector order: base, interfaces, tip. For two pneumatic segments, the connector length array is `[base, interface_0_1, tip]`.

```python
params = ISupportParams(
    ...,
    length=jnp.array([190e-3, 180e-3]),
    pcs_segment_lengths=jnp.array([95e-3, 95e-3, 60e-3, 60e-3, 60e-3]),
    rigid_connector_lengths=jnp.array([41e-3, 27e-3, 6e-3]),
)
structure = ISupportStructure(
    pcs_segment_counts=(2, 3),
    rigid_connector_selector=(True, True, True),
)
robot = ISupport(params=params, structure=structure)
```

Updating metric parameters keeps the topology fixed. If explicit PCS lengths are used, update them consistently with pneumatic segment lengths.

```python
updated = robot.update_params(
    length=jnp.array([200e-3, 180e-3]),
    pcs_segment_lengths=jnp.array([100e-3, 100e-3, 60e-3, 60e-3, 60e-3]),
    rigid_connector_lengths=jnp.array([41e-3, 27e-3, 8e-3]),
)
```

## Example Updates

- `examples/simulation/pcs/simulate_isupport.py` now models two pneumatic segments with lengths 190 mm and 180 mm.
- Rigid connectors are included in the main example: 41 mm base, 27 mm interface, and 6 mm tip.
- The example exposes `pcs_segment_counts`, `pcs_segment_lengths`, `rigid_connector_selector`, and `rigid_connector_lengths` so users can try different discretizations.
- The base frame and gravity are oriented so the robot points along the world-frame negative z-axis.
- The old rigid-interface-only example was removed in favor of the main I-SUPPORT example.

## Tests

- `uv run ruff check src/soromox/systems/params.py src/soromox/systems/pcs/structures.py src/soromox/systems/pcs/params.py src/soromox/systems/pcs/isupport.py examples/simulation/pcs/simulate_isupport.py tests/systems/test_pressure_actuated_pcs_models.py`
- `uv run pytest tests/systems/test_pressure_actuated_pcs_models.py tests/systems/test_system_lengths.py`
- `uv run python -m py_compile src/soromox/systems/params.py src/soromox/systems/pcs/structures.py src/soromox/systems/pcs/params.py src/soromox/systems/pcs/isupport.py examples/simulation/pcs/simulate_isupport.py`

### CODE REVIEW

**Review by Elektron97 — APPROVED** [review]
Everything good! In case I'll add the damping matrix with the classic PCS/GVS formulation. Maybe we can do a selectable flag for that.
